tests/Hal/JsonSchema/SchemaFactoryTest.php

```diff
@@ -0,0 +1,129 @@
+use ApiPlatform\Core\Hal\JsonSchema\SchemaFactory;
+use ApiPlatform\Core\Hydra\JsonSchema\SchemaFactory as HydraSchemaFactory;
+use ApiPlatform\Core\JsonSchema\Schema;
+use ApiPlatform\Core\JsonSchema\SchemaFactory as BaseSchemaFactory;
+use ApiPlatform\Core\JsonSchema\TypeFactoryInterface;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyNameCollectionFactoryInterface;
+use ApiPlatform\Core\Metadata\Property\PropertyNameCollection;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\Dummy;
+use ApiPlatform\Core\Tests\ProphecyTrait;
+use PHPUnit\Framework\TestCase;
+
+class SchemaFactoryTest extends TestCase
+{
+    use ProphecyTrait;
+
+    private $schemaFactory;
+
+    protected function setUp(): void
+    {
+        $typeFactory = $this->prophesize(TypeFactoryInterface::class);
+        $resourceMetadataFactory = $this->prophesize(ResourceMetadataFactoryInterface::class);
+        $resourceMetadataFactory->create(Dummy::class)->willReturn(new ResourceMetadata(Dummy::class));
+        $propertyNameCollectionFactory = $this->prophesize(PropertyNameCollectionFactoryInterface::class);
+        $propertyNameCollectionFactory->create(Dummy::class, ['enable_getter_setter_extraction' => true])->willReturn(new PropertyNameCollection());
+        $propertyMetadataFactory = $this->prophesize(PropertyMetadataFactoryInterface::class);
+
+        $baseSchemaFactory = new BaseSchemaFactory(
+            $typeFactory->reveal(),
+            $resourceMetadataFactory->reveal(),
+            $propertyNameCollectionFactory->reveal(),
+            $propertyMetadataFactory->reveal()
+        );
+
+        $hydraSchemaFactory = new HydraSchemaFactory($baseSchemaFactory);
+
+        $this->schemaFactory = new SchemaFactory($hydraSchemaFactory);
+    }
+
+    public function testBuildSchema(): void
+    {
+        $resultSchema = $this->schemaFactory->buildSchema(Dummy::class);
+
+        $this->assertTrue($resultSchema->isDefined());
+        $this->assertEquals(str_replace('\\', '.', Dummy::class).'.jsonhal', $resultSchema->getRootDefinitionKey());
+    }
+
+    public function testCustomFormatBuildSchema(): void
+    {
+        $resultSchema = $this->schemaFactory->buildSchema(Dummy::class, 'json');
+
+        $this->assertTrue($resultSchema->isDefined());
+        $this->assertEquals(str_replace('\\', '.', Dummy::class), $resultSchema->getRootDefinitionKey());
+    }
+
+    public function testHasRootDefinitionKeyBuildSchema(): void
+    {
+        $resultSchema = $this->schemaFactory->buildSchema(Dummy::class);
+        $definitions = $resultSchema->getDefinitions();
+        $rootDefinitionKey = $resultSchema->getRootDefinitionKey();
+
+        $this->assertEquals(str_replace('\\', '.', Dummy::class).'.jsonhal', $rootDefinitionKey);
```

Comment:
This line duplicates a previous test.

Reply:
I removed the duplicate assert also from the hydra test.